shm/src/Item.cpp

```diff
@@ -0,0 +1,48 @@
+#include "shm/inc/Item.hpp"
+
+Item::Item(const std::string& name, size_t amount, size_t basePrice, Rarity rarity)
+    : Cargo(name, amount, basePrice), rarity_(rarity)
+{}
+
+bool Item::operator==(const Cargo& cargo) const {
+    if (typeid(cargo) == typeid(Item)) {
+        const Item* item = static_cast<const Item*>(&cargo);
+        return name_ == item->getName() && amount_ == item->getAmount() && basePrice_ == item->getBasePrice() &&
+               rarity_ == item->getRarity();
+    }
+
+    return false;
+}
+
+Cargo& Item::operator+=(size_t amount) {
+    amount_ += amount;
+
+    return *this;
+}
+
+Cargo& Item::operator-=(size_t amount) {
+    if (amount <= amount_) {
+        amount_ -= amount;
+    }
+
+    return *this;
```

Comment:
Redundant line

Reply:
newline-before-return rule:
This rule requires an empty line before return statements to increase code clarity, except when the return is alone inside a statement group (such as an if statement). In the latter case, the return statement does not need to be delineated by virtue of it being alone.